.travis.yml

```diff
@@ -22,16 +27,34 @@ env:
    #   via the "travis encrypt" command using the project repo's public key
    - secure: "R/JfoBMrkhCGWhfWM1m3gPHuLtMBlp2SIK1R9BaPbRsbGBUJmAg9V0g0YXSaw8SVxoyuiL/jsLtHPfDeub9oTxrYydew+6/4KaoQdG7EGXQJfBhH2f0ag/hTKJfXnmZX9jMMnTxPf5Axjq+w4E6sKkU2+d1oAJRhrqzYNwDhVlc="
    - CXX_STD: "c++11"
-   - LCS_ENGINE_FLAGS: "-ui"
+
+# Install any required tools
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]] ; then sudo gem install xcpretty ; fi
 
 # Set up the source tree by fetching Linux-specific prebuilt objects
-install: (cd prebuilt && ./fetch-libraries.sh linux)
+install: 
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]] ; then (cd prebuilt && ./fetch-libraries.sh linux) ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx"   ]] ; then (cd prebuilt && ./fetch-libraries.sh mac)   ; fi
 
 # Bootstrap the LCB compiler, build the default target set and run a
 # the default test suite.
 script: >
   if [ "${COVERITY_SCAN_BRANCH}" != "1" ]; then
-      make all-linux && make check-linux
+    if [[ "$TRAVIS_OS_NAME" == "linux" ]] ; then
+      LICENSE_FILE=livecode-firstrun.lcf
+      LICENSE_DIR="$HOME/.runrev/licenses"
+      [ -d "$LICENSE_DIR" ] || mkdir -p "$LICENSE_DIR"
+      touch "$LICENSE_DIR/$LICENSE_FILE"
+      make all-linux && xvfb-run make check-linux
+    fi
+    if [[ "$TRAVIS_OS_NAME" == "osx" ]] ; then
+      LICENSE_FILE=livecode-firstrun.lcf
+      LICENSE_DIR="$HOME/Library/Application Support/RunRev/Licenses"
+      [ -d "$LICENSE_DIR" ] || mkdir -p "$LICENSE_DIR"
+      touch "$LICENSE_DIR/$LICENSE_FILE"
+      make all-mac && make check-mac
+    fi
```

Comment:
It would be great to make the license file stuff as shared as possible. I wonder if something like this would work:

```sh
set -eu
case "${TRAVIS_OS_NAME}" in
  linux)
    BUILD_PLATFORM=linux
    LICENSE_DIR="${HOME}/.runrev/licenses"
    ;;
  osx)
    BUILD_PLATFORM=mac
    LICENSE_DIR="${HOME}/Library/Application Support/RunRev/Licenses
    export XCODEBUILD="set -o pipefail && xcodebuild"
    export XCODEBUILD_FILTER="| xcpretty"
    ;;
esac

if [ "${COVERITY_SCAN_BRANCH}" != "1" ]; then
  mkdir -p "${LICENSE_DIR}"
  touch "$LICENSE_DIR/livecode-firstrun.lcf"

  make all-${BUILD_PLATFORM}
  make check-${BUILD_PLATFORM}
fi
```